includes/class-wc-gateway-swedbank-pay-checkout.php

```diff
@@ -0,0 +1,1837 @@
+		$params = [
+			'operation'           => 'initiate-consumer-session',
+			'consumerCountryCode' => $country,
+		];
+
+		try {
+			$result = $this->request( 'POST', '/psp/consumers', $params );
+			$js_url = self::get_operation( $result['operations'], 'view-consumer-identification' );
+			wp_send_json_success( $js_url );
+		} catch ( Exception $e ) {
+			wp_send_json_error( $e->getMessage() );
+		}
+	}
+
+	/**
+	 * Ajax: Retrieve Consumer Profile Reference
+	 *
+	 * @return void
+	 */
+	public function ajax_swedbank_pay_checkout_customer_profile() {
+		check_ajax_referer( 'swedbank_pay_checkout', 'nonce' );
+
+		$customer_reference = isset( $_POST['consumerProfileRef'] ) ? wc_clean( $_POST['consumerProfileRef'] ) : '';
+		if ( empty( $customer_reference ) ) {
+			wp_send_json_error( [ 'message' => 'Customer reference required' ] );
+			exit();
+		}
+
+		// Store Customer Profile
+		$url = WC()->session->get( 'consumer_js_url' );
+		$this->update_consumer_profile( get_current_user_id(),  $customer_reference, $url );
+
+		wp_send_json_success();
+	}
+
+	/**
+	 * Ajax: Place Order.
+	 *
+	 * @throws Exception
+	 */
+	public function ajax_swedbank_pay_place_order() {
+		check_ajax_referer( 'swedbank_pay_checkout', 'nonce' );
+
+		$data = [];
+		parse_str( $_POST['data'], $data );
+		$_POST = $data;
+		unset( $_POST['terms-field'], $_POST['terms'] );
+
+		$_POST['payment_method'] = 'payex_checkout';
+
+		$_REQUEST['woocommerce-process-checkout-nonce'] = wp_create_nonce( 'woocommerce-process_checkout' );
+		$_POST['_wpnonce']                              = wp_create_nonce( 'woocommerce-process_checkout' );
+
+		WC()->checkout()->process_checkout();
+	}
+
+	/**
+	 * Ajax: Update Order's data.
+	 *
+	 * @throws Exception
+	 */
+	public function ajax_swedbank_pay_update_order() {
+		check_ajax_referer( 'swedbank_pay_checkout', 'nonce' );
+
+		// Get Order
+		$order_id = absint( WC()->session->get( 'order_awaiting_payment' ) );
+		if ( $order_id === 0 ) {
+			wp_send_json( [
+				'result'   => 'failure',
+				'messages' => 'Order is not exists',
+			] );
+
+			return;
+		}
+
+		// Get Order
+		$order = wc_get_order( $order_id );
+		if ( ! $order ) {
+			wp_send_json( [
+				'result'   => 'failure',
+				'messages' => 'Order is not exists',
+			] );
+
+			return;
+		}
+
+		// Mark order failed instead of cancelled
+		if ( $order->get_payment_method() === $this->id && $order->has_status( 'cancelled' ) ) {
+			$order->update_status( 'failed' );
+		}
+
+		// Prepare $_POST data
+		$data = [];
+		parse_str( $_POST['data'], $data );
+		$_POST = $data;
+		unset( $_POST['terms-field'], $_POST['terms'] );
+
+		$_POST['payment_method'] = $this->id;
+
+		if ( ! empty( $_POST['compat'] ) && (bool) $_POST['compat'] ) {
+			$_POST['is_update_backward_compat'] = 1;
+		} else {
+			$_POST['is_update'] = '1';
+		}
+
+		// Update Checkout
+		// @see WC_AJAX::update_order_review()
+		if ( ! empty( $_POST['shipping_method'] ) && ! is_array( $_POST['shipping_method'] ) ) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Direct use of $_POST Superglobal detected.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)